packages/relay/src/lib/clients/sdkClient.ts

```diff
@@ -498,75 +503,125 @@ export class SDKClient {
     }
   }
 
+  /**
+   * Executes transaction -> handle errors -> capture burned HBAR in metrics and HBAR rate limit class.
+   * @param transaction
+   * @param callerName
+   * @param interactingEntity
+   * @param requestId
+   * @returns
+   */
   async executeTransaction(
     transaction: Transaction,
     callerName: string,
     interactingEntity: string,
     requestId: string,
-  ): Promise<TransactionResponse> {
+  ): Promise<TransactionResponse | TransactionResponse[]> {
     const transactionType = transaction.constructor.name;
```

Comment:
Are we sure this would work? Have you tested it? 

Btw, why not just do `transaction instanceof FileAppendTransaction` instead of checking the constructor name?

Reply:
Yeah we've been using this `transaction.constructor.name` for quite a while. This is to get the type of the transaction for the purposes of logging and adding to metrics. 

Though I like this `transaction instanceof FileAppendTransaction` for checking if transactions are FileAppend transactions. But now we're splitting the executeTransaction so it might not bee necessary.